db/system_keyspace.cc

```diff
@@ -2636,6 +2651,299 @@ class clients_table : public streaming_virtual_table {
+        _db(db), _ss(ss), _gossiper(gossiper) {}
+};
+
+class keyspace_elements_description_table : public streaming_virtual_table {
+    replica::database& _db;
+
+    static schema_ptr build_schema() {
+        using desc = cql3::statements::describe_statement;
+        auto id = generate_legacy_id(system_keyspace::NAME, desc::ELEMENTS_TABLE);
+        return schema_builder(system_keyspace::NAME, desc::ELEMENTS_TABLE, std::make_optional(id))
+            .with_column(to_bytes(desc::KEYSPACE_COLUMN), utf8_type, column_kind::partition_key)
+            .with_column(to_bytes(desc::TYPE_COLUMN), utf8_type, column_kind::clustering_key)
+            .with_column(to_bytes(desc::NAME_COLUMN), utf8_type, column_kind::clustering_key)
+            .with_column(to_bytes(desc::CREATE_COLUMN), utf8_type)
+            .with_column(to_bytes(desc::CREATE_WITH_INTERNALS_COLUMN), utf8_type)
+            .with_version(system_keyspace::generate_schema_version(id))
+            .build();
+    }
+
+    dht::decorated_key make_partition_key(const sstring& value) {
+        return dht::decorate_key(*_s, partition_key::from_single_value(*_s, data_value(value).serialize_nonnull()));
+    }
+
+    clustering_key make_clustering_key(const sstring& type, const sstring& name) {
+        return clustering_key::from_exploded( *_s, {
+            data_value(type).serialize_nonnull(),
+            data_value(name).serialize_nonnull()
+        });
+    }
+
+    bool is_index(const table_id& id, const schema_ptr& schema) {
+        return _db.find_column_family(id).get_index_manager().is_index(*schema);
+    }
+
+    future<> emit_row(result_collector& result, const sstring& type, const sstring& name, const sstring& details, const sstring& details_with_internals) {
+        clustering_row cr(make_clustering_key(type, name));
+        set_cell(cr.cells(), to_bytes(cql3::statements::describe_statement::CREATE_COLUMN), details);
+        set_cell(cr.cells(), to_bytes(cql3::statements::describe_statement::CREATE_WITH_INTERNALS_COLUMN), details_with_internals);
+        co_await result.emit_row(std::move(cr));
+    }
+
+    template <typename E>
+    sstring get_descriptions(const E& element) {
+        std::ostringstream os;
+
+        element.describe(os);
+        return os.str();
+    }
+
+    future<> aggregates(result_collector& result, const sstring& ks) {
+        auto udas = cql3::functions::functions::get_user_aggregates(ks);
+        boost::sort(udas, [] (const auto& a, const auto& b) {
+            return a->name().name < b->name().name;
+        });
+
+        for (auto& uda : udas) {
+            auto desc = get_descriptions(*uda);
+            co_await emit_row(result, "aggregate", cql3::util::maybe_quote(uda->name().name), desc, desc);
+        }
+    }
+
+    future<> functions(result_collector& result, const sstring& ks) {
```

Comment:
Don't name functions as nouns (unless it's a getter). Functions do something. What does co_await function(x, y) do? nobody can guess.

Reply:
In the rewritten version, I've left the names like (function(), functions(), ...) but their signatures have changed and they act as getters to descriptions (also described in comment in the code).